tests/integration/helpers.py

```diff
@@ -65,3 +67,18 @@ async def get_unit_address(ops_test: OpsTest, app_name: str, unit_no: int):
     if unit is None:
         assert False, f"no unit exists in app {app_name} with index {unit_no}"
     return unit["address"]
+
+
+@retry(wait=wait_fixed(10), stop=stop_after_attempt(10))
+async def check_agent_data_in_mimir(ops_test: OpsTest, coordinator_app: str) -> Dict[str, Any]:
```

Comment:
not a blocker, but I'm trying to make a habit of adding a comment (or a docstring) to explain what the test tests, unless obvious. In this case I'd have liked one. 
Also, adding something like: GIVEN/WHEN/THEN comments to spell out what you're doing and checking.

Reply:
Added more documentation!